project/src/pages/player-page/player-page.tsx

```diff
@@ -1,42 +1,45 @@
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
+import { useNavigate } from 'react-router-dom';
 import Spinner from '../../components/spinner/spinner';
-import { isActiveFilmLoaded } from '../../constants/constants';
+import { FilmType } from '../../types/films-type';
+
+type PlayerPageProps = {
+  film: FilmType,
+}
+
+function PlayerPage({film}: PlayerPageProps): JSX.Element {
+  const {previewVideoLink, previewImage} = film;
+
+  const navigate = useNavigate();
+  const [isLoading, setIsLoading] = useState(true);
+  const videoRef = useRef<HTMLVideoElement | null>(null);
+
+  useEffect(() => {
+    if (videoRef.current === null) {
+      return;
+    }
+
+    videoRef.current.addEventListener('loadeddata', () => setIsLoading(false));
+
+    videoRef.current.play();
+    videoRef.current.defaultPlaybackRate = 1.0;
+    videoRef.current.controls = true;
+
+  }, [film.id]);
 
-function PlayerPage() {
   return (
     <div className="player">
-      {isActiveFilmLoaded
-        ? <video src="#" className="player__video" poster="img/player-poster.jpg"></video>
-        : <Spinner />}
-
-      <button type="button" className="player__exit">Exit</button>
-
-      <div className="player__controls">
-        <div className="player__controls-row">
-          <div className="player__time">
-            <progress className="player__progress" value="30" max="100"></progress>
-            <div className="player__toggler" style={{left: '30%'}}>Toggler</div>
-          </div>
-          <div className="player__time-value">1:30:29</div>
-        </div>
-
-        <div className="player__controls-row">
-          <button type="button" className="player__play">
-            <svg viewBox="0 0 19 19" width="19" height="19">
-              <use xlinkHref="#play-s"></use>
-            </svg>
-            <span>Play</span>
-          </button>
-          <div className="player__name">Transpotting</div>
-
-          <button type="button" className="player__full-screen">
-            <svg viewBox="0 0 27 27" width="27" height="27">
-              <use xlinkHref="#full-screen"></use>
-            </svg>
-            <span>Full screen</span>
-          </button>
-        </div>
-      </div>
+      {isLoading ? <Spinner /> : ''}
+
+      <video
+        className="player__video"
+        poster={previewImage}
+        src={previewVideoLink}
+        ref={videoRef}
+      >
+      </video>
+
+      <button type="button" className="player__exit" onClick={() => navigate(-1)}>Exit</button>
```

Comment:
```suggestion
      <button 
      type="button" 
      className="player__exit"
       onClick={() => navigate(-1)}
       >
         Exit
       </button>
```

самой не сложно читать код?